shared/chat/conversation/info-panel/attachments.tsx

```diff
@@ -0,0 +1,577 @@
+import * as React from 'react'
+import * as Types from '../../../constants/types/chat2'
+import * as RPCChatTypes from '../../../constants/types/rpc-chat-gen'
+import * as Kb from '../../../common-adapters'
+import * as Styles from '../../../styles'
+import {imgMaxWidthRaw} from '../messages/attachment/image/image-render'
+import {formatTimeForMessages} from '../../../util/timestamp'
+import MessagePopup from '../messages/message-popup'
+// @ts-ignore
+import {Props as HeaderHocProps} from '../../../common-adapters/header-hoc/types'
+
+const monthNames = [
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December',
+]
+
+const getDateInfo = thumb => {
+  const date = new Date(thumb.ctime)
+  return {
+    month: monthNames[date.getMonth()],
+    year: date.getFullYear(),
+  }
+}
+
+const formMonths = thumbs => {
```

Comment:
I think this function and usage is pretty confusing. Especially the _finalizeMonths below which mutate their function arguments.

Not sure the best fix here, but a thought:

* Have this function take the renderSection, renderItem, and mapData functions so that you can remove usages of `_finalizeMonths` I get that you want to split up logic, but in this case _finalizeMonths implicitly knows the structure of a month. Probably better to keep this in one place.

Reply:
Ok changed some stuff around, hopefully more clear, and with more types.